proposals/ytt/003-openapi-schemas/README.md

```diff
@@ -0,0 +1,85 @@
+# Generate OpenAPI Schema
+
+- Status: **Scoping** | Pre-Alpha | In Alpha | In Beta | GA | Rejected
+- Originating Issue: [ytt#103](https://github.com/vmware-tanzu/carvel-ytt/issues/103)
+
+# Problem Statement
+Configuration Authors want to be able to generate documentation for configuration inputs (for example, Data Values Schema, and Data Values) for their users. Additionally, Configuration Consumers want to be able to validate their configuration inputs by other tools (for example, IDE's and OpenAPI Schema validators). 
+
+# Proposal
+Implement a flag to create an [OpenAPI Schema](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md) yaml document generated from a [Data Values schema](https://carvel.dev/ytt/docs/latest/lang-ref-ytt-schema/) file, and an optional Data Values file. When invoking ytt with configuration inputs and the flag `--schema-inspect-openapi`, the result is the OpenAPI formatted Schema object only. 
+In addition, the `--schema-inspect-openapi-metadata` flag may be used to include _both_ the [OpenAPI metadata](#openapi-document-with-metadata) and the [Schema object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#schema-object).
```

Comment:
@danielhelfand Are you suggesting that (a) the OpenAPI metadata is unnecessary, (b) that the OpenAPI metadata should always be returned, or (c) something else?

Reply:
I am referring to just the name of the flag (i.e. `--schema-inspect-openapi-metadata`). It's already a lengthy name and to me the word metadata does not make the purpose much clearer. I think `--schema-inspect-openapi` would be clearer, but I am guessing metadata was added since the generation flag is named `--schema-inspect-openapi`. My [comment](https://github.com/vmware-tanzu/carvel/pull/198#discussion_r685266514) above mentions that I think that flag name should be something different.

Reply:
In fact, when I see the name `--...openapi-metadata`, I would presume that it would include _only_ the metadata (i.e. API name and Info object).

When I move my frame of reference over to the OpenAPI world and look back at `ytt`, then I would assume that the default output is a proper/complete OpenAPI Document (which is exactly what's in the proposal). And from that same perspective, limiting the output to the Schema component feels like a flavor-of the OpenAPI Document.

A corresponding (evolved) set of flags is detailed in [this other comment](https://github.com/vmware-tanzu/carvel/pull/198#discussion_r687197686).

Reply:
Great points! ❤️  The flag name has been changed in the updated proposal.